spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -384,10 +390,9 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
 				consumer.assign(new ArrayList<>(this.definedPartitions.keySet()));
 			}
 			this.consumer = consumer;
-			Object theListener = listener == null ? ackListener : listener;
 			GenericErrorHandler<?> errHandler = this.containerProperties.getGenericErrorHandler();
-			if (theListener instanceof MessageListener) {
-				this.listener = (MessageListener<K, V>) theListener;
+			if (this.theListener instanceof MessageListener) {
+				this.listener = (MessageListener<K, V>) this.theListener;
 				this.batchListener = null;
```

Comment:
As I see, this code is a part of constructor. In this case initializing object's field with `null` is unnecessary - java does this. So now it's just extra code, which makes the method longer and less readable.